compiler/test/suites/provides.re

```diff
@@ -181,4 +181,19 @@ describe("provides", ({test, testSkip}) => {
     "module Test; provide let rec foo: () => Number = () => 5",
     ("foo", Binaryen.Export.external_function),
   );
+  assertHasWasmExport(
+    "issue_1884_type_provided_later1",
+    "module Test; enum T { A }; let a = A; provide { type T }; provide { a }",
+    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
+  );
+  assertHasWasmExport(
+    "issue_1884_type_provided_later2",
+    "module Test; enum T { A }; let a = A; provide { a, type T }",
+    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
+  );
+  assertHasWasmExport(
+    "issue_1884_type_provided_later3",
+    "module Test; enum T { A }; let a = A; provide { a }; provide { type T }",
+    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
+  );
```

Comment:
```suggestion
  );
   assertHasWasmExport(
    "issue_1884_type_provided_later3",
    "module Test; enum T { A }; provide let a = A; provide { type T }",
    ("GRAIN$EXPORT$a", Binaryen.Export.external_global),
  );
```
This is likely a more common way that you'd provide a value and the type after. I assume it works, but let's add a test to be certain